cli/command_user_hash_password_test.go

```diff
@@ -0,0 +1,76 @@
+package cli_test
+
+import (
+	"math/rand"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/kopia/kopia/internal/testutil"
+	"github.com/kopia/kopia/tests/testenv"
+)
+
+func TestServerUserHashPassword(t *testing.T) {
+	const (
+		userName = "user78"
+		userHost = "client-host"
+		userFull = userName + "@" + userHost
+	)
+
+	runner := testenv.NewInProcRunner(t)
+	e := testenv.NewCLITest(t, testenv.RepoFormatNotImportant, runner)
+
+	e.RunAndExpectSuccess(t, "repo", "create", "filesystem", "--path", e.RepoDir, "--override-username", "server", "--override-hostname", "host")
+
+	t.Cleanup(func() {
+		e.RunAndExpectSuccess(t, "repo", "disconnect")
+	})
+
+	userPassword := "bad-password-" + strconv.Itoa(int(rand.Int31()))
+
+	out := e.RunAndExpectSuccess(t, "server", "users", "hash-password", "--user-password", userPassword)
+
+	require.Len(t, out, 1)
+
+	passwordHash := out[0]
+	require.NotEmpty(t, passwordHash)
+
+	// attempt to create a user with a bad password hash
+	e.RunAndExpectFailure(t, "server", "users", "add", userFull, "--user-password-hash", "bad-base64")
+
+	// create a new user with and set the password using the password hash
+	e.RunAndExpectSuccess(t, "server", "users", "add", userFull, "--user-password-hash", passwordHash)
```

Comment:
Do we want to test `server users set` also? Perhaps we can update the password hash and show fail/success when the same user connects with old/new passwords?

Reply:
@redgoat650 PTAL.
I added a success case after changing the user password (via a hash).
I'll owe the failure cases, those surfaced a minor issue with a resource not being released when the client command fails to execute. It may be similar to the issue in #3980 but on the client side. 3980 is on the server side, it was surfaced when fixing a test.